### ISSUE TITLE
signature mismatch: mark wrong declaration instead of whole module?

### ISSUE THREAD

**Issue by vogler** [hackwaly/vscode-ocaml#38]
``` ocaml
module type S = sig
    type t
    val f1: t -> t
    val f2: t -> t
end

module M : S = struct
    type t = ()
    let f1 () = 1
    let f2 () = ()
end
```

Here only `f1` is problematic but the whole `struct ... end` is underlined in red.
It would be helpful if we could use this information from the error message

```
File "Untitled-1", line 3, characters 4-18: Expected declaration
File "Untitled-1", line 9, characters 8-10: Actual declaration
```

in order to highlight the first declaration that doesn't match.

**Comment by hackwaly** [comment]
Nice to have.

**Comment by hackwaly** [comment]
This is released in v0.5.0. Enjoy!
